### PR TITLE
Add support for expanded SUBTABLE

### PR DESCRIPTION
**Pull request by t3t5u**
Add support for expanded SUBTABLE.

Process flow:
1. Write records to a temporary file as CSV from each task thread.
2. Merge temporary files from each task into a single temporary file.
3. Sort a merged temporary file.
4. Read records from a sorted temporary file as CSV.
5. Reduce records to SUBTABLE rows.

Normal sequence:
- `KintonePageOutput` > `PageReader` > `KintoneColumnVisitor`

Reduce sequence:
- `ReducedPageOutput` > `PageReader` > `CSVOutputColumnVisitor` >
- `CSVPrinter` > Temporary file per each task thread >
- `Reducer` > Merged into a single temporary file > Sorted temporary file >
- `CSVParser` > Reduced to SUBTABLE rows > `CSVInputColumnVisitor` > `PageBuilder` >
- `KintonePageOutput` > `PageReader` > `KintoneColumnVisitor`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +32,10 @@ kintone output plugin for Embulk stores app records from kintone.
         - `USER_SELECT`, `ORGANIZATION_SELECT`, `GROUP_SELECT`, `FILE`
     - **timezone**: timezone to convert into `date` (string, default is `UTC`)
     - **val_sep**: Used to specify multiple checkbox values (string, default is `,`)
+    - **sort_columns**: List of columns for sorting rows in SUBTABLE. Available only if type is `SUBTABLE` (array of objects, optional)
+        - **name**: Column name (string, required)
+        - **order**: Sort order (string `asc` or `desc`, required)
+- **chunk_size**: Maximum number of records to request at once (integer, default is `100`)
```
Comment (t3t5u) [inline README.md:38]: To avoid the following error.
https://jp.cybozu.help/k/en/id/040458.html#anchor_filter_records_error

**Comment by chikamura** [comment]
@t3t5u 
Could you please add some examples to ## Example section of the README.md?

**Comment by t3t5u** [comment]
@chikamura 

I've added simplest example for reduce expanded SUBTABLE to README.
https://github.com/trocco-io/embulk-output-kintone/pull/38/commits/fcf0e53103a358461cff6b1af944011eb4655b1d

For a more complete example, please refer to the [config.yml](https://github.com/trocco-io/embulk-output-kintone/pull/38/files#diff-893769150c66ad4d0d728e86f437443b06ee83ce237f6bf2d5f6ded40aa80190) for test.

File: src/main/java/org/embulk/output/kintone/reducer/ReduceType.java [hunk 2]
```diff
@@ -141,7 +141,7 @@ public static MapValue value(Value value) {
 
   public static MapValue value(Long id, MapValue value, MapValue sortValue) {
     ValueFactory.MapBuilder builder = ValueFactory.newMapBuilder();
-    builder.put(ID, id == null ? NIL : ValueFactory.newInteger(id));
+    builder.put(ID, id == null ? NIL : ValueFactory.newString(id.toString()));
```
Comment (t3t5u) [inline src/main/java/org/embulk/output/kintone/reducer/ReduceType.java:144]: SUBTABLE row id is Long in [kintone-java-client](https://github.com/kintone/kintone-java-client/blob/master/src/main/java/com/kintone/client/model/record/TableRow.java#L20), but it is string in [API document](https://cybozu.dev/ja/kintone/docs/overview/field-types/#field-type-update).

File: README.md [hunk 3]
```diff
@@ -46,6 +56,19 @@ out:
     date_time: {field_code: "datetime", type: "DATETIME"}
 ```
```
Comment (chikamura) [inline README.md:58]: Could you add a supplement, as in the example below?


#### KINTONE

| Form                      | Field Code      |
| ------------------------- | --------------- |
| Table's own               | table           |
| NUMBER In Table           | number_in_table |
| SINGLE_LINE_TEXT In Table | text_in_table   |

#### INPUT CSV

```csv
id,table.number,table.text
1,0,test0
1,1,test1
2,0,test0
```

#### RESULT

ID:1

| NUMBER        | SINGLE_LINE_TEXT |
| ------------- | ---------------- |
| 0             | test0            |
| 1             | test1            |

ID:2

| NUMBER        | SINGLE_LINE_TEXT |
| ------------- | ---------------- |
| 0             | test0            |

File: src/main/java/org/embulk/output/kintone/reducer/Reducer.java
Comment (chikamura) [inline src/main/java/org/embulk/output/kintone/reducer/Reducer.java:1]: nits: Could the generic var or func name be changed to a more descriptive name?

#### example
indices -> notReducedIndices
range -> tableRootRange
size ->  ...
schema -> ...

**Review by chikamura — APPROVED** [review]

**Review by d-hrs — APPROVED** [review]
LGTM

**Pull request merged**